### PR TITLE
Metrics summary quantiles are incorrect.

### PR DESCRIPTION
**Pull request by tbjuhasz**
Quantiles are inaccurate when only a few samples are present. This was particularly visible when testing for quartiles. The solution updated the way the rank was calculated by ensuring that the data is divided evenly across all data points.

### CODE REVIEW

File: src/ddsketch.rs [hunk 1]
```diff
@@ -222,25 +222,108 @@ mod tests {
...
             assert_relative_eq!(dd.quantile(q).unwrap().unwrap(), val, max_relative = alpha);
         }
     }
+    #[test]
+    fn test_negative_quartiles() {
+        let alpha = 0.01;
+        let c = Config::new(alpha, 2048, 10e-9);
+        let mut dd = DDSketch::new(c);
+
+        // Initialize sketch with {1.0, 2.0, 3.0, 4.0}
```
Comment (dfwilbanks395) [inline src/ddsketch.rs:247]: Update. Or remove since the comment's not really necessary. Can be updated in a few places below too

**Review by dwlbnks — COMMENTED** [review]
lgtm overall, but I'll leave approval to @mheffner.

**Comment by mheffner** [comment]
Thanks @tbjuhasz and @dfwilbanks395 for taking a look. I'll review shortly and get this in.

**Comment by mheffner** [comment]
Hey @tbjuhasz, sorry for the delay here. So this library was originally modeled after the Go [version](https://github.com/DataDog/sketches-go) of the same. Granted, I haven't been able to stay on top of this so the Go version has had significant improvements since I originally wrote this port.

However, I wanted to double check the behavior between the two libraries to make sure they are at least producing the same results. I plugged in the values from the test suites and compared the resulting quantiles with the Go lib. AFAICT, the quantiles removed from the test here are matching what the Go lib produces, but I'm not getting the same test results with the new quantiles added (0.26, 0.51, 0.76). I'd like to better understand why we are no longer aligning with those results. Here is the Go test app: https://gist.github.com/mheffner/6434273d60cfe156505066e6211b38b5

Thoughts? I'm curious if you think the behavior of the Go library is correct or not.

**Comment by tbjuhasz** [comment]
Hello @mheffner, We are testing the quarters and in that case 1,2,3 and 4 should all end up in their own quarter. But the current implementation does not due this. This problem shows up when there only a few data points.

**Comment by mheffner** [comment]
> Hello @mheffner, We are testing the quarters and in that case 1,2,3 and 4 should all end up in their own quarter. But the current implementation does not due this. This problem shows up when there only a few data points.

Have you discussed this with the Datadog team? I'm curious what their thoughts are on the right approach here. As I mentioned, I'm trying to mostly keep this in-sync with the Go version of this implementation to avoid drift. Maybe there's a way to make this a configurable approach?

**Comment by mheffner** [comment]
I appreciate the work here @tbjuhasz, but I'm afraid I'll have to close this. This breaks the desire for this library to maintain compatible behavior with the [Go version](https://github.com/DataDog/sketches-go). I'd be happy to revisit this it if that assumption changes in the future though.

**Pull request closed without merging**